content/en/docs/armory-agent/agent-mtls.md

```diff
@@ -0,0 +1,267 @@
+
+```bash
+kubectl create secret generic <agent-secret-name> \
+--from-file=agent.crt --from-file=agent.key
+```
+
+### Modify deployment configuration
+
+Modify the Agent's deployment configuration to mount the certs. @TODO what file is this in?
+
+If you have a custom CA, you need to mount the cert into the known trusted cert location:  `/etc/ssl/cert.pem`. Golang apps in Alpine use `/etc/ssl/cert.pem` as the source of a trusted CA.
+
+  1. Obtain a copy of the `cert.pem` file from a running Agent.
+  2. Append your custom CA onto the copied `cert.pem`  (for example `cat ca.crt >> cert.pem`)
+  3. Mount `cert.pem` to Agent deployment
+
+@TODO is this config correct? do the "this didn't work" sections work now or not?
+
+{{< prism lang="yaml"  >}}
+spec:
+  template:
+    spec:
+      containers:
+        volumeMounts:
+        - mountPath: /opt/kubesvc/cert
+          name: armoryagentcert
+        - mountPath: /opt/spinnaker/config
+          name: volume-kubesvc-config
+        - mountPath: /opt/kubesvc/cacert  # this didn't work as of Nov 2020
+          name: clouddrivercacert         # this didn't work as of Nov 2020
+      volumes:
+      - name: armoryagentcert
+        secret:
+          secretName: <agent-secret-name>
+      - name: clouddrivercacert         # this didn't work as of Nov 2020
+        secret:                         # this didn't work as of Nov 2020
+          secretName: <your-secret-name> # this didn't work as of Nov 2020
+      - name: certpem
+        secret:
+          secretName: ??? @TODO what secret is this?
+{{< /prism >}}
+
+
+### Configure the service
+
+@TODO what file is this in?
+
+{{< prism lang="yaml" >}}
+clouddriver:
+  grpc: <:443
+  insecure: false
+  tls:
+    #serverName: my-ca  #to override the server name to verify
+    insecureSkipVerify: false #if true, don't verify server's cert
+    clientKeyFile: /opt/kubesvc/cert/agent.key #ref to the private key (mTLS)
+    #clientKeyPassword: #if the clientKeyFile is password protected
+    #cacertFile: /opt/kubesvc/cacert/ca.crt #to validate server's cert
+    clientCertFile: /opt/kubesvc/cert/agent.crt #client cert for mTLS.
+
+{{< /prism >}}
+
+See the {{< linkWithTitle "agent-options.md" >}} for additional options.
+
+## Forward proxy configuration
+
+### Add proxy variables
+
+Add proxy variables in the Agent's deployment file. @TODO what file is this?
+
+{{< prism lang="yaml" line="25-30" >}}
+spec:
+  template:
+    spec:
+      containers:
+      - image: armory/kubesvc
+        env:
+        - name: HTTP_PROXY
+          value: <proxyaddress:proxyport>
+        - name: HTTPS_PROXY
+          value: <proxyaddress:proxyport>
+{{< /prism >}}
+
+### Configure the Agent
+
+Configure the Agent to **not** proxy Kubernetes traffic intended for KubeAPI in your cluster.
+
+Add the `noProxy: true` option to have traffic destined for Kubernetes ignore the proxy environment settings under `kubernetes`.
+
+@TODO what file is this?
```

Comment:
According to https://docs.armory.io/docs/armory-agent/agent-options/ this would be in the `kubesvc.yaml` configuration file.